frontend/src/Metamaps/Views/ContextMenu.js

```diff
@@ -0,0 +1,108 @@
+import Control from '../Control'
+import DataModel from '../DataModel'
+import { ReactApp } from '../GlobalUI'
+import Selected from '../Selected'
+import Topic from '../Topic'
+
+const ContextMenu = {
+  clickedNode: null,
+  clickedEdge: null,
+  pos: {x: 0, y: 0},
+  fetchingSiblingsData: false,
+  siblingsData: null,
+  selectNode: (node, pos) => {
+    ContextMenu.pos = pos
+    ContextMenu.clickedNode = node
+    ContextMenu.clickedEdge = null
+    ContextMenu.fetchingSiblingsData = false
+    ContextMenu.siblingsData = null
+    ReactApp.render()
```

Comment:
this makes me nervous. Something about accessing global objects in this way. The Control stuff doesn't so much, because it's hard to avoid.

Could we pass a render callback to each of these functions that calls ReactApp.render? I think that would be an improvement. At least it would make these more functional/testable

Reply:
I'm not sure if I should ask about including tests for some of these files or not ;)